packages/@aws-cdk/aws-batch/lib/managed-compute-environment.ts

```diff
@@ -0,0 +1,1222 @@
+   * @default 30 minutes
+   */
+  readonly updateTimeout?: Duration;
+
+  /**
+   * Whether or not any running jobs will be immediately terminated when an infrastructure update
+   * occurs. If this is enabled, any terminated jobs may be retried, depending on the job's
+   * retry policy.
+   *
+   * @see https://docs.aws.amazon.com/batch/latest/userguide/updating-compute-environments.html
+   *
+   * @default false
+   */
+  readonly terminateOnUpdate?: boolean;
+
+  /**
+   * The security groups this Compute Environment will launch instances in.
+   */
+  readonly securityGroups: ec2.ISecurityGroup[];
+
+  /**
+   * The VPC Subnets this Compute Environment will launch instances in.
+   */
+  readonly vpcSubnets?: ec2.SubnetSelection;
+
+  /**
+   * Whether or not the AMI is updated to the latest one supported by Batch
+   * when an infrastructure update occurs.
+   *
+   * If you specify a specific AMI, this property will be ignored.
+   *
+   * @default true
+   */
+  readonly updateToLatestImageVersion?: boolean;
+}
+
+/**
+ * Props for a ManagedComputeEnvironment
+ */
+export interface ManagedComputeEnvironmentProps extends ComputeEnvironmentProps {
+  /**
+  * The maximum vCpus this `ManagedComputeEnvironment` can scale up to.
+  * Each vCPU is equivalent to 1024 CPU shares.
+  *
+  * *Note*: if this Compute Environment uses EC2 resources (not Fargate) with either `AllocationStrategy.BEST_FIT_PROGRESSIVE` or
+  * `AllocationStrategy.SPOT_CAPACITY_OPTIMIZED`, or `AllocationStrategy.BEST_FIT` with Spot instances,
+  * The scheduler may exceed this number by at most one of the instances specified in `instanceTypes`
+  * or `instanceClasses`.
+  *
+  * @default 256
+  */
+  readonly maxvCpus?: number;
+
+  /**
+  * Specifies whether this Compute Environment is replaced if an update is made that requires
+  * replacing its instances. To enable more properties to be updated,
+  * set this property to `false`. When changing the value of this property to false,
+  * do not change any other properties at the same time.
+  * If other properties are changed at the same time,
+  * and the change needs to be rolled back but it can't,
+  * it's possible for the stack to go into the UPDATE_ROLLBACK_FAILED state.
+  * You can't update a stack that is in the UPDATE_ROLLBACK_FAILED state.
+  * However, if you can continue to roll it back,
+  * you can return the stack to its original settings and then try to update it again.
+  *
+  * The properties which require a replacement of the Compute Environment are:
+  *
+  * `allocationStrategy`, `spotBidPercentage`, // TODO
+  * @see: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-batch-computeenvironment.html#cfn-batch-computeenvironment-replacecomputeenvironment
+  * @see: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-continueupdaterollback.html
+  *
+  * @default true
```

Comment:
Batch may want this value to be `false` by default for new CFN based CEs since replacement is the legacy behavior once Batch released dynamic infra updates in CEs, https://docs.aws.amazon.com/batch/latest/userguide/updating-compute-environments.html

I will follow up once I have more info, FYI for now...

Reply:
after syncing with some Batch SMEs, we concluded that `ReplaceComputeEnvironment` should default to `false` because that is the better experience long-term for ComputeEnvironments created by these new constructs.